crates/red_knot/src/types/infer.rs

```diff
@@ -148,6 +148,15 @@ fn infer_expr_type(db: &dyn SemanticDb, file_id: FileId, expr: &ast::Expr) -> Qu
             let attr_name = &Name::new(&attr.id);
             value_type.get_member(db, attr_name)
         }
+        ast::Expr::Call(ast::ExprCall { func, .. }) => {
+            let func_type = infer_expr_type(db, file_id, func)?;
+            // TODO: check that func_type is a function type, else type error? are we doing
+            // type-checking here?
```

Comment:
I think our approach in inference-only mode in cases that should really be a type error should be to substitute `Unknown` type.